src/Compilers/CSharp/Portable/Symbols/Synthesized/InjectedNonNullTypesAttributeSymbol.cs

```diff
@@ -0,0 +1,133 @@
+            if (ctor is null)
+            {
+                // member is missing
+                return ImmutableArray<CSharpAttributeData>.Empty;
+            }
+
+            Binder.ReportUseSiteDiagnostics(ctor, _diagnostics, Location.None);
+
+            NamedTypeSymbol attributeTargets = DeclaringCompilation.GetWellKnownType(WellKnownType.System_AttributeTargets);
+            Binder.ReportUseSiteDiagnostics(attributeTargets, _diagnostics, Location.None);
+
+            var usage = new SynthesizedAttributeData(ctor,
+                arguments: ImmutableArray.Create(new TypedConstant(attributeTargets, TypedConstantKind.Enum, attributeUsage)),
+                namedArguments: ImmutableArray<KeyValuePair<string, TypedConstant>>.Empty);
+
+            return ImmutableArray.Create<CSharpAttributeData>(usage);
+        }
+
+        internal override AttributeUsageInfo GetAttributeUsageInfo()
+            => new AttributeUsageInfo(validTargets: attributeUsage, allowMultiple: false, inherited: false);
+
+        private const AttributeTargets attributeUsage =
+            AttributeTargets.Class |
+            AttributeTargets.Constructor |
+            AttributeTargets.Delegate |
+            AttributeTargets.Enum |
+            AttributeTargets.Event |
+            AttributeTargets.Field |
+            AttributeTargets.Interface |
+            AttributeTargets.Method |
+            AttributeTargets.Module |
+            AttributeTargets.Property |
+            AttributeTargets.Struct;
+
+        private sealed class NonNullTypesAttributeConstructorSymbol : SynthesizedInstanceConstructor
+        {
+            private readonly ImmutableArray<ParameterSymbol> _parameters;
+
+            internal NonNullTypesAttributeConstructorSymbol(
+                NamedTypeSymbol containingType,
+                Func<MethodSymbol, ImmutableArray<ParameterSymbol>> getParameters) :
+                base(containingType)
+            {
+                Debug.Assert(containingType is InjectedNonNullTypesAttributeSymbol);
+                _parameters = getParameters(this);
+            }
+
+            public override ImmutableArray<ParameterSymbol> Parameters
+                => _parameters;
+
+            internal override bool SynthesizesLoweredBoundBody
+                => true;
+
+            /// <summary>
+            /// Note: this method captures diagnostics into the containing type (an injected attribute symbol) instead,
+            /// as we don't yet know if the containing type will be emitted.
+            /// </summary>
+            internal override void GenerateMethodBody(TypeCompilationState compilationState, DiagnosticBag diagnostics)
+            {
+                var containingType = (InjectedNonNullTypesAttributeSymbol)ContainingType;
+                GenerateMethodBodyCore(compilationState, containingType._diagnostics);
```

Comment:
>GenerateMethodBodyCore(compilationState, containingType._diagnostics); [](start = 16, length = 70)

Let's assume this call produces diagnostics. Every time we try to emit, will we add it again? So, each subsequent attempt will have more and more duplicates?